python/packages/ai/teams/ai/augmentations/tools_augmentation.py

```diff
@@ -0,0 +1,100 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from __future__ import annotations
+
+import json
+from typing import List, Optional
+
+from botbuilder.core import TurnContext
+
+from ...state import MemoryBase
+from ..models.chat_completion_action import ChatCompletionAction
+from ..models.prompt_response import PromptResponse
+from ..planners.plan import (
+    Plan,
+    PredictedCommand,
+    PredictedDoCommand,
+    PredictedSayCommand,
+)
+from ..prompts.sections.prompt_section import PromptSection
+from ..tokenizers.tokenizer import Tokenizer
+from ..validators.validation import Validation
+from .augmentation import Augmentation
+
+
+class ToolsAugmentation(Augmentation[str]):
+    """
+    A server-side 'tools' augmentation.
+    """
+
+    _actions: Optional[List[ChatCompletionAction]]
+
+    def __init__(self, actions: Optional[List[ChatCompletionAction]]) -> None:
+        self._actions = actions
+
+    def create_prompt_section(self) -> Optional[PromptSection]:
+        """
+        Creates an optional prompt section for the augmentation.
+        """
+        return None
+
+    async def validate_response(
+        self,
+        context: TurnContext,
+        memory: MemoryBase,
+        tokenizer: Tokenizer,
+        response: PromptResponse[str],
+        remaining_attempts: int,
+    ) -> Validation:
+        """
+        Validates a response to a prompt.
+
+        Args:
+            context (TurnContext): Context for the current turn of conversation.
+            memory (MemoryBase): Interface for accessing state variables.
+            tokenizer (Tokenizer): Tokenizer to use for encoding/decoding text.
+            response (PromptResponse[str]): Response to validate.
+            remaining_attempts (int): Nubmer of remaining attempts to validate the response.
+
+        Returns:
+            Validation: A 'Validation' object.
+        """
+        return Validation(valid=True)
+
+    async def create_plan_from_response(
+        self,
+        turn_context: TurnContext,
+        memory: MemoryBase,
```

Comment:
Is there a reason for using `MemoryBase` instead of `Memory`? In .ts, I see the augmentation files using `Memory`.